spec/system/admin_badges_spec.rb

```diff
@@ -63,6 +63,39 @@
     end
   end
 
+  context "when updating a badge" do
+    it "can upload an image for the badge" do
+      badges_page.visit_page(Badge::Autobiographer)
+      form.choose_conditional("upload-image")
+
+      attach_file(File.absolute_path(file_from_fixtures("logo.jpg"))) do
+        form.field("image_url").find(".image-upload-controls .btn").click
+      end
+
+      expect(form.field("image_url")).to have_css(".btn-danger")
+      form.submit
+
+      expect(PageObjects::Components::Toasts.new).to have_success(I18n.t("js.saved"))
+      badge = Badge.find(Badge::Autobiographer)
+      expect(badge.image_upload_id).to be_present
+      expect(badge.icon).to be_blank
```

Comment:
Might want to wrap this in a `try_until_sucess` block since we don't actually know when the server might finish processing the request and save the image into the database.